snomed/com.b2international.snowowl.snomed.datastore.server/src/com/b2international/snowowl/datastore/server/snomed/index/InitialReasonerTaxonomyBuilder.java

```diff
@@ -509,42 +431,54 @@ public InitialReasonerTaxonomyBuilder(final RevisionSearcher searcher, final Typ
 				fullyDefinedConceptIdsReference, 
 				SnomedConceptDocument.Expressions.defining());
 
-		ForkJoinUtils.runInParallel(
-				taxonomyBuilderRunnable,
-				getConceptConcreteDomainsRunnable,
-				getInferredConceptConcreteDomainsRunnable,
-				getStatementConcreteDomainsRunnable,
-				getExhaustiveConceptIdsRunnable,
-				getFullyDefinedConceptIdsRunnable);
-
-		exhaustiveConceptIds = exhaustiveConceptIdsReference.get();
-		fullyDefinedConceptIds = fullyDefinedConceptIdsReference.get();
-		conceptIdToConcreteDomain = conceptConcreteDomainReference.get();
-		inferredConcreteDomainMap = inferredConceptConcreteDomainReference.get();
-		statementIdToConcreteDomain = relationshipConcreteDomainReference.get();
-
-		for (final LongIterator itr = conceptIdToConcreteDomain.keySet().iterator(); itr.hasNext(); /* empty */) {
-			final long conceptId = itr.next();
-			final Collection<ConcreteDomainFragment> fragments = getConceptConcreteDomainFragments(conceptId);
-
-			for (final ConcreteDomainFragment fragment : fragments) {
-				synchronized (storageKeyLock) {
-					componentStorageKeyToConceptId.put(fragment.getStorageKey(), conceptId);
-				}
+		if (concreteDomainSupportEnabled) {
+			final AtomicReference<LongKeyMap<Collection<ConcreteDomainFragment>>> conceptConcreteDomainReference = createAtomicReference();
+			final AtomicReference<LongKeyMap<Collection<ConcreteDomainFragment>>> inferredConceptConcreteDomainReference = createAtomicReference();
+			final AtomicReference<LongKeyMap<Collection<ConcreteDomainFragment>>> relationshipConcreteDomainReference = createAtomicReference();
+			
+			final Collection<String> allowedCharacteristicTypes = getAllowedCharacteristicTypes();
+			// XXX: Change processor mode needs additional concrete domain members as well (relationships only)
+			if (!isReasonerMode()) {
```

Comment:
This flag is only ever `false` if the "strength" index of SNOMED CT pharma products needs to be updated in a commit change processor.